File: python/pad_array.py
def pad(array, min_size, value=None):
    """
    This function returns a new array padded with the value

    Args:
        array (list): the array to pad
        min_size (int): the minimum size of the new array
        value (any): the value to pad with

    Returns:
        list: the padded array
    """

    if min_size <= len(array):
        return array

    return array + [value] * (min_size - len(array))

File: js/padArray.js
/**
 * This function returns the given array padded with the given fill element until
 * the array has the given minimum size.
 *
 * @param {Array} array The array to pad.
 * @param {Integer} minSize The minimum size of the array.
 * @param {Any} value The value to pad with.
 * @returns {Array} The padded array.
 */
const pad = (array, minSize, value = null) => {
  if (minSize <= array.length) return array;

  return array.concat(Array(minSize - array.length).fill(value));
};

module.exports = { pad };
